src/components/TextField.tsx

```diff
@@ -94,43 +107,66 @@ class LinodeTextField extends React.Component<CombinedProps> {
     );
   }
 
+  handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { type, min, max, onChange } = this.props;
+
+    const numberTypes = ['tel', 'number'];
+
+    /** because !!0 is falsy :( */
+    const minAndMaxExist = typeof min === 'number' && typeof max === 'number';
+
+    /**
+     * if we've provided a mix and max value, make sure the user
+     * input doesn't go outside of those bounds ONLY if the input
+     * type matches a number type
+     */
+    const cleanedValue =
+      minAndMaxExist && numberTypes.some(eachType => eachType === type)
```

Comment:
these changes here were part of a PR that already got merged to develop. I was just copy/pasting